packages/language-core/lib/codegen/script/scriptSetup.ts

```diff
@@ -323,6 +325,9 @@ function* generateComponentProps(
 			if (defineProp.type) {
 				yield scriptSetup.content.substring(defineProp.type.start, defineProp.type.end);
 			}
+			else if (defineProp.name && defineProp.nameIsString) {
+				yield `NonNullable<typeof ${propName}['value']>`;
+			}
 			else if (!defineProp.nameIsString) {
```

Comment:
Consider merging with this code branch?

Reply:
What does this usecase actually cover? `!defineProp.nameIsString`

Reply:
Merged branches

Reply:
Not sure tho 😂